conans/paths/package_layouts/package_cache_layout.py

```diff
@@ -136,3 +161,29 @@ def package_summary_hash(self, pref):
         package_folder = self.package(pref)
         readed_manifest = FileTreeManifest.load(package_folder)
         return readed_manifest.summary_hash
+
+    # Locks
+    def conanfile_read_lock(self, output):
+        if self._no_lock:
+            return NoLock()
+        return ReadLock(self.conan(), self._ref, output)
+
+    def conanfile_write_lock(self, output):
+        if self._no_lock:
+            return NoLock()
+        return WriteLock(self.conan(), self._ref, output)
+
+    def conanfile_lock_files(self, output):
+        if self._no_lock:
+            return ()
```

Comment:
not sure, is it expected here? it differs in return value from other functions, may be add comment, if it's really required to return an empty tuple

Reply:
Yes, it is, it is returning an iterable of _lock files_, and it is used elsewhere in a loop: `for f in package_layout.conanfile_lock_files(output=output):`, so we need it to return an iterable.